t/live-https.t fails with IO::Socket::SSL 1.49

```
$ cpanm --local-lib local-lib --reinstall Net::HTTP -v
cpanm (App::cpanminus) 1.7042 on perl 5.012004 built for darwin-thread-multi-2level
Work directory is /Users/al/.cpanm/work/1493464230.49632
You have make /usr/bin/make
You have LWP 6.03
You have /usr/bin/tar: bsdtar 2.8.3 - libarchive 2.8.3
You have /usr/bin/unzip
Searching Net::HTTP () on cpanmetadb ...
--> Working on Net::HTTP
Fetching http://www.cpan.org/authors/id/O/OA/OALDERS/Net-HTTP-6.14.tar.gz ... OK
Unpacking Net-HTTP-6.14.tar.gz
x Net-HTTP-6.14/
x Net-HTTP-6.14/Changes
x Net-HTTP-6.14/CONTRIBUTORS
x Net-HTTP-6.14/cpanfile
x Net-HTTP-6.14/dist.ini
x Net-HTTP-6.14/INSTALL
x Net-HTTP-6.14/lib/
x Net-HTTP-6.14/LICENSE
x Net-HTTP-6.14/Makefile.PL
x Net-HTTP-6.14/MANIFEST
x Net-HTTP-6.14/META.json
x Net-HTTP-6.14/META.yml
x Net-HTTP-6.14/perlcriticrc
x Net-HTTP-6.14/perltidyrc
x Net-HTTP-6.14/README.md
x Net-HTTP-6.14/t/
x Net-HTTP-6.14/tidyall.ini
x Net-HTTP-6.14/t/00-report-prereqs.dd
x Net-HTTP-6.14/t/00-report-prereqs.t
x Net-HTTP-6.14/t/http-nb.t
x Net-HTTP-6.14/t/http.t
x Net-HTTP-6.14/t/live-https.t
x Net-HTTP-6.14/t/live.t
x Net-HTTP-6.14/t/rt-112313.t
x Net-HTTP-6.14/lib/Net/
x Net-HTTP-6.14/lib/Net/HTTP/
x Net-HTTP-6.14/lib/Net/HTTP.pm
x Net-HTTP-6.14/lib/Net/HTTPS.pm
x Net-HTTP-6.14/lib/Net/HTTP/Methods.pm
x Net-HTTP-6.14/lib/Net/HTTP/NB.pm
Entering Net-HTTP-6.14
Checking configure dependencies from META.json
Checking if you have ExtUtils::MakeMaker 6.58 ... Yes (7.24)
Running Makefile.PL
Configuring Net-HTTP-6.14 ... Checking if your kit is complete...
Looks good
Generating a Unix-style Makefile
Writing Makefile for Net::HTTP
Writing MYMETA.yml and MYMETA.json
OK
Checking dependencies from MYMETA.json ...
Checking if you have warnings 0 ... Yes (1.09)
Checking if you have base 0 ... Yes (2.15)
Checking if you have URI 0 ... Yes (1.59)
Checking if you have IO::Uncompress::Gunzip 0 ... Yes (2.024)
Checking if you have Data::Dumper 0 ... Yes (2.125)
Checking if you have ExtUtils::MakeMaker 0 ... Yes (7.24)
Checking if you have File::Spec 0 ... Yes (3.62)
Checking if you have IO::Socket::INET 0 ... Yes (1.31)
Checking if you have vars 0 ... Yes (1.01)
Checking if you have Compress::Raw::Zlib 0 ... Yes (2.024)
Checking if you have Carp 0 ... Yes (1.38)
Checking if you have strict 0 ... Yes (1.04)
Checking if you have Test::More 0 ... Yes (1.302078)
Checking if you have IO::Select 0 ... Yes (1.17)
Checking if you have Socket 0 ... Yes (2.020)
Building and testing Net-HTTP-6.14 ... cp lib/Net/HTTP/NB.pm blib/lib/Net/HTTP/NB.pm
cp lib/Net/HTTP.pm blib/lib/Net/HTTP.pm
cp lib/Net/HTTPS.pm blib/lib/Net/HTTPS.pm
cp lib/Net/HTTP/Methods.pm blib/lib/Net/HTTP/Methods.pm
Manifying 4 pod documents
PERL_DL_NONLAZY=1 "/usr/bin/perl" "-MExtUtils::Command::MM" "-MTest::Harness" "-e" "undef *Test::Harness::Switches; test_harness(0, 'blib/lib', 'blib/arch')" t/*.t
t/00-report-prereqs.t .. #
# Versions for all modules listed in MYMETA.json (including optional ones):
#
# === Configure Requires ===
#
#     Module              Want Have
#     ------------------- ---- ----
#     ExtUtils::MakeMaker  any 7.24
#
# === Configure Suggests ===
#
#     Module      Want    Have
#     -------- ------- -------
#     JSON::PP 2.27300 2.27203
#
# === Build Requires ===
#
#     Module              Want Have
#     ------------------- ---- ----
#     ExtUtils::MakeMaker  any 7.24
#
# === Test Requires ===
#
#     Module              Want     Have
#     ------------------- ---- --------
#     Data::Dumper         any    2.125
#     ExtUtils::MakeMaker  any     7.24
#     File::Spec           any     3.62
#     IO::Select           any     1.17
#     Socket               any    2.020
#     Test::More           any 1.302078
#
# === Test Recommends ===
#
#     Module         Want     Have
#     ---------- -------- --------
#     CPAN::Meta 2.120900 2.150005
#
# === Runtime Requires ===
#
#     Module                 Want  Have
#     ---------------------- ---- -----
#     Carp                    any  1.38
#     Compress::Raw::Zlib     any 2.024
#     IO::Socket::INET        any  1.31
#     IO::Uncompress::Gunzip  any 2.024
#     URI                     any  1.59
#     base                    any  2.15
#     strict                  any  1.04
#     vars                    any  1.01
#     warnings                any  1.09
#
# === Runtime Suggests ===
#
#     Module            Want Have
#     ----------------- ---- ----
#     IO::Socket         any 1.31
#     IO::Socket::INET6  any 2.67
#     IO::Socket::IP     any 0.37
#     IO::Socket::SSL   1.38 1.49
#     Symbol             any 1.07
#
t/00-report-prereqs.t .. ok
t/http-nb.t ............ ok
t/http.t ............... ok
t/live-https.t ......... Net::HTTPS: SSL connect attempt failed with unknown errorerror:1407742E:SSL routines:SSL23_GET_SERVER_HELLO:tlsv1 alert protocol version at t/live-https.t line 34.
# Looks like your test exited with 255 before it could output anything.
t/live-https.t ......... Dubious, test returned 255 (wstat 65280, 0xff00)
Failed 6/6 subtests
t/live.t ............... ok
t/rt-112313.t .......... ok

Test Summary Report
-------------------
t/live-https.t       (Wstat: 65280 Tests: 0 Failed: 0)
  Non-zero exit status: 255
  Parse errors: Bad plan.  You planned 6 tests but ran 0.
Files=6, Tests=98, 28 wallclock secs ( 0.06 usr  0.02 sys +  0.88 cusr  0.14 csys =  1.10 CPU)
Result: FAIL
Failed 1/6 test programs. 0/98 subtests failed.
make: *** [test_dynamic] Error 255
FAIL
! Installing Net::HTTP failed. See /Users/al/.cpanm/work/1493464230.49632/build.log for details. Retry with --force to force install it.
```

IO::Socket::SSL is 1.49.
Upgrading to 2.048 fixed the issue. You might want to raise the dependency version.

--- a/Changes
+++ b/Changes
@@ -1,6 +1,7 @@
 Release history for {{$dist->name}}
 
 {{$NEXT}}
+    - Bump IO::Socket::SSL version from 1.38 to 2.012
 
 6.15      2017-05-12 14:57:02+02:00 Europe/Paris
     - Fix t/rt-112313.t (Shoichi Kaji)

--- a/META.json
+++ b/META.json
@@ -46,7 +46,7 @@
             "IO::Socket" : "0",
             "IO::Socket::INET6" : "0",
             "IO::Socket::IP" : "0",
-            "IO::Socket::SSL" : "1.38",
+            "IO::Socket::SSL" : "2.012",
             "Symbol" : "0"
          }
       },

--- a/cpanfile
+++ b/cpanfile
@@ -11,7 +11,7 @@ requires "warnings" => "0";
 suggests "IO::Socket" => "0";
 suggests "IO::Socket::INET6" => "0";
 suggests "IO::Socket::IP" => "0";
-suggests "IO::Socket::SSL" => "1.38";
+suggests "IO::Socket::SSL" => "2.012";
 suggests "Symbol" => "0";
 
 on 'test' => sub {

--- a/dist.ini
+++ b/dist.ini
@@ -29,7 +29,7 @@ perl = 5.006002
 skip = Net::SSL
 
 [Prereqs / RuntimeSuggests]
-IO::Socket::SSL = 1.38
+IO::Socket::SSL = 2.012
 
 [Prereqs::Soften]
 to_relationship = suggests

